### DISCUSSION TITLE
Determine whether there are consumers in the queue

### DISCUSSION THREAD

**Discussion by wdqzdd** [rabbitmq/rabbitmq-dotnet-client#1074] [General]
I want to realize single consumption，But my application is clustered，then there are many consumers

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
Depending on the intent you can either [monitor the number of consumers](https://www.rabbitmq.com/monitoring.html) or make the target queue only allow for a [Single Active Consumer](https://rabbitmq.com/consumers.html#single-active-consumer).

**Reply by wdqzdd** [reply]
thank you,I solved this problem by getting the number of consumers.

 channel.QueueDeclarePassive(queueName).ConsumerCount

**Comment by wdqzdd** [comment]
thank you
